internal/kibana/data_view/schema_test.go

```diff
@@ -0,0 +1,329 @@
+					ID:            "id",
+					TimeFieldName: utils.Pointer("time_field_name"),
+					SourceFilters: []string{"field1", "field2"},
+					FieldAttributes: map[string]apiFieldAttrsV0{
+						"field1": {
+							CustomLabel: utils.Pointer("custom_label"),
+							Count:       utils.Pointer[int64](10),
+						},
+					},
+					RuntimeFieldMap: map[string]apiRuntimeFieldV0{
+						"runtime_field": {
+							Type:         "keyword",
+							ScriptSource: "emit(\"hello\")",
+						},
+					},
+					FieldFormats: map[string]apiFieldFormat{
+						"field1": {
+							ID: "field1",
+							Params: &apiFieldFormatParams{
+								Pattern: "0.00",
+							},
+						},
+					},
+					AllowNoIndex: true,
+					Namespaces:   []string{"default", "o11y"},
+				},
+			},
+			expectedRequest: data_views.CreateDataViewRequestObject{
+				Override: utils.Pointer(false),
+				DataView: data_views.CreateDataViewRequestObjectDataView{
+					AllowNoIndex: utils.Pointer(true),
+					FieldAttrs: map[string]interface{}{
+						"field1": fieldAttr{
+							CustomLabel: utils.Pointer("custom_label"),
+							Count:       utils.Pointer[int64](10),
+						},
+					},
+					FieldFormats: map[string]interface{}{
+						"field1": apiFieldFormat{
+							ID: "field1",
+							Params: &apiFieldFormatParams{
+								Pattern: "0.00",
+							},
+						},
+					},
+					Id:         utils.Pointer("id"),
+					Name:       utils.Pointer("name"),
+					Namespaces: []string{"default", "o11y"},
+					RuntimeFieldMap: map[string]interface{}{
+						"runtime_field": runtimeField{
+							Type: "keyword",
+							Script: runtimeFieldSource{
+								Source: "emit(\"hello\")",
+							},
+						},
+					},
+					SourceFilters: []data_views.SourcefiltersInner{
+						{Value: "field1"},
+						{Value: "field2"},
+					},
+					TimeFieldName: utils.Pointer("time_field_name"),
+					Title:         "title",
+				},
+			},
+		},
+		{
+			name: "nil collections",
+			model: apiModelV0{
+				DataView: apiDataViewV0{
+					Title:         utils.Pointer("title"),
+					Name:          utils.Pointer("name"),
+					ID:            "id",
+					TimeFieldName: utils.Pointer("time_field_name"),
+					AllowNoIndex:  true,
+				},
+			},
+			expectedRequest: data_views.CreateDataViewRequestObject{
+				Override: utils.Pointer(false),
+				DataView: data_views.CreateDataViewRequestObjectDataView{
+					AllowNoIndex:  utils.Pointer(true),
+					Id:            utils.Pointer("id"),
+					Name:          utils.Pointer("name"),
+					TimeFieldName: utils.Pointer("time_field_name"),
+					Title:         "title",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var tfModel tfModelV0
+			diags := tfsdk.ValueFrom(context.Background(), tt.model, getSchema().Type(), &tfModel)
+			require.Nil(t, diags)
+
+			req, diags := tfModel.ToCreateRequest(context.Background())
+
+			require.Equal(t, tt.expectedRequest, req)
+			require.Equal(t, tt.expectedDiags, diags)
```

Comment:
Probably looking at this wrong, but do we not want to test cases where this request returns diags? I don't see `expectedDiags` in any of the test cases.

Reply:
It would be nice to test some of the type conversions. It's quite difficult to do so, TF doesn't make it easy to define nested TF objects which is what we'd need to do to test the error cases here.